### DISCUSSION TITLE
Skip_dir and sync_dir help

### DISCUSSION THREAD

**Discussion by rnpereira** [abraunegg/onedrive#1527] [Q&A] [answered]
Hello,

When I Installed onedrive I choose the default settings and it installed on my /home. My linux is sharing a SSD with windows, but I have a 1TB HDD to put files. I forgot to point the HDD for the OneDrive folder, and also that I had a 60 gb photo folder on the cloud.
So, when I hit synced ondrive, all of it came to my SSD, and now I have only 20gb left of space.

I changed the config file and added the folder I dont want to be synced, and also changed the syn_dir

In the config file I have:
> sync_dir = "/mnt/7693df7e-ee0b-40a2-94dd-0d79a04a77df/OneDrive/"
 skip_dir = "SkyDrive\ camera\ roll"

And here is my config display:
```
onedrive version                       = v2.4.10-0~202102221319~ubuntu20.04.1
Config path                            = /home/ricardo/.config/onedrive
Config file found in config path       = true
Config option 'check_nosync'           = false
Config option 'sync_dir'               = /mnt/7693df7e-ee0b-40a2-94dd-0d79a04a77df/OneDrive/
Config option 'skip_dir'               = SkyDrive\ camera\ roll
`Config option 'skip_file'              = ~*|.~*|*.tmp`
Config option 'skip_dotfiles'          = false
Config option 'skip_symlinks'          = false
Config option 'monitor_interval'       = 300
Config option 'min_notify_changes'     = 5
Config option 'log_dir'                = /var/log/onedrive/
Config option 'classify_as_big_delete' = 1000
Config option 'upload_only'            = false
Config option 'no_remote_delete'       = false
Config option 'remove_source_files'    = false
Config option 'sync_root_files'        = false
Selective sync 'sync_list' configured  = false
Business Shared Folders configured     = false
```
Then I ran the sync and got that:
```
onedrive --dry-run --synchronize --resync 
Configuration file successfully loaded
config file has been updated, checking if --resync needed
DRY-RUN Configured. Output below shows what 'would' have occurred.
Configuring Global Azure AD Endpoints
Initializing the Synchronization Engine ...
Syncing changes from OneDrive ...
Creating local directory: SkyDrive camera roll
```
I expected to not sync the SkiDrive folder, but it seems that  it is synchronizing.

Am I missing something? Did I spelled something wrong? Besides the skip_dir problem, does it look ok to change the directory to from my ssd to my hdd?

I've installed onedrive through apt-get.

Thank you

**Comment by abraunegg** [comment, marked as answer]
@rnpereira 
>  Besides the skip_dir problem, does it look ok to change the directory to from my ssd to my hdd?

If your `/mnt/7693df7e-ee0b-40a2-94dd-0d79a04a77df/OneDrive` path is the correct location for your Windows data then yes, just be sure to *not* enable 'files on demand' in Windows otherwise that would lead to major issues for you.

Regarding your 'skip_dir' issue, you are adding '\ ' as part of your configuration - this means, it is trying to match a path that has '\ ' in it - which is not what you want / need to do.

You need to remove that and use just `skip_dir = "SkyDrive camera roll"` - refer to the examples as per: https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#skip_dir

> I've installed onedrive through apt-get.

You are running an unsupported client version - refer to https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md#installing-from-distribution-packages and upgrade your client.

**Reply by rnpereira** [reply]
@abraunegg 

>If your /mnt/7693df7e-ee0b-40a2-94dd-0d79a04a77df/OneDrive path is the correct location for your Windows data then yes, just be sure to not enable 'files on demand' in Windows otherwise that would lead to major issues for you.

Yes, this is the path for my secondary hdd. But this is a partition for Linux only. It is not sharing space with Windows.

>Regarding your 'skip_dir' issue, you are adding '\ ' as part of your configuration - this means, it is trying to match a path that has '\ ' in it - which is not what you want / need to do.
You need to remove that and use just skip_dir = "SkyDrive camera roll" - refer to the examples as per: https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#skip_dir

Thank you! I thought I had to add \ because the space in the folder name. I ran the command again and now the SkiDrive didnt synched.

>You are running an unsupported client version - refer to https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md#installing-from-distribution-packages and upgrade your client.

You mean install directly from git? Because the Ubuntu PPA still have the 2.4.10. Do I have to uninstall the version I have?

One more question, if I change the sync_dir, it will move the files that I already have here, or it will download everything again? Either way, can I delete the SkiDrive folder after that, or it will do it automatically? 

Thank you.

**Comment by abraunegg** [comment]
@rnpereira 
> You mean install directly from git? Because the Ubuntu PPA still have the 2.4.10. Do I have to uninstall the version I have?

By using the PPA you are using an unsupported version. You also are impacted by this issue: https://github.com/abraunegg/onedrive/issues/1274

You need to uninstall the PPA, ensure all files are removed, then compile from source.

> One more question, if I change the sync_dir, it will move the files that I already have here, or it will download everything again? 

When you update the 'sync_dir' option, you will need to ensure that you first stop the client so that there is no active running process (see above bug with PPA). Once the application is stopped, move all your files to your new location, reconfigure the client, then restart the client.

>  Either way, can I delete the SkiDrive folder after that, or it will do it automatically?

It is best to manually delete this from all locations where it is not required, as, now you are skipping it, deleting it will not have any 'sync' effect. It is best to perform all your clean up operations when you are re-configuring / installing the correct client version.
